core/src/main/scala/org/apache/spark/metrics/MetricsSystem.scala

```diff
@@ -133,7 +133,12 @@ private[spark] class MetricsSystem private (
 
     if (instance == "driver" || instance == "executor") {
       if (metricsNamespace.isDefined && executorId.isDefined) {
-        MetricRegistry.name(metricsNamespace.get, executorId.get, source.sourceName)
+        if (source.sourceName.contains("TIBCO ComputeDB") ||
+            source.sourceName.contains("SnappyData")) {
+          MetricRegistry.name("", "", source.sourceName)
+        } else {
+          MetricRegistry.name(metricsNamespace.get, executorId.get, source.sourceName)
```

Comment:
Why else is required. If for Spark side metrics then please add a comment. Also please put a comment as to why you are ignoring the metricsNamespace and executorId.

Reply:
Added suggested changes